Modix.Data/Repositories/PromotionDialogRepository.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Modix.Data.Models.Promotions;
+
+namespace Modix.Data.Repositories
+{
+    public interface IPromotionDialogRepository
+    {
+        /// <summary>
+        /// Creates a new dialog entity and adds it to the database
+        /// </summary>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        Task CreateAsync(ulong messageId, long campaignId);
+
+        /// <summary>
+        /// gets the current active dialogs
+        /// </summary>
+        /// <returns></returns>
+        Task<IReadOnlyList<PromotionDialogBrief>> GetDialogsAsync();
+
+        /// <summary>
+        /// Attempts to delete a poll by its message id
+        /// </summary>
+        /// <param name="messageId"></param>
+        /// <returns></returns>
+        Task<bool> TryDeleteAsync(ulong messageId);
+    }
+
+    public class PromotionDialogRepository : RepositoryBase, IPromotionDialogRepository
+    {
+        public PromotionDialogRepository(ModixContext modixContext) : base(modixContext) { }
+
+        public async Task CreateAsync(ulong messageId, long campaignId)
+        {
+            await ModixContext.Set<PromotionDialogEntity>().AddAsync(new PromotionDialogEntity
```

Comment:
It's actually recommended to use `Add` instead of `AddAsync` when you don't use special value generators that need to access the database. More info: https://docs.microsoft.com/dotnet/api/microsoft.entityframeworkcore.dbset-1.addasync?view=efcore-3.1

```suggestion
            await ModixContext.Set<PromotionDialogEntity>().Add(new PromotionDialogEntity
```

Reply:
Does it still need to be awaited?

Reply:
Ah, you're right. Nope, no need to await `Add`